### ISSUE TITLE
Add setting to change output name of ScalaJS bundle

### ISSUE THREAD

**Issue by gehnaphore** [com-lihaoyi/mill#496]
The filename is important because the generated source map refers to it by that name.

**Comment by lefou** [comment]
Is this still valid? Don't we support something like modules which affect the output names? /cc @lolgab

**Comment by lihaoyi** [comment]
This is now possible via `def scalaJSOutputPatterns = OutputPatterns.fromJSFile("foo")`

```scala

object foo extends ScalaJSModule {
  def scalaVersion = "2.13.8"
  def scalaJSVersion = "1.13.0"
  def ivyDeps = Agg(ivy"com.lihaoyi::scalatags::0.12.0")

  def scalaJSOutputPatterns = OutputPatterns.fromJSFile("foo")
}
```

**Comment by lefou** [comment]
* Fixed by #2233
